uni/lib/view/Widgets/generic_expansion_card.dart

```diff
@@ -0,0 +1,40 @@
+import 'package:flutter/material.dart';
+import 'package:expansion_tile_card/expansion_tile_card.dart';
+
+abstract class GenericExpansionCard extends StatefulWidget {
+  const GenericExpansionCard({Key? key}) : super(key: key);
+
+  @override
+  State<StatefulWidget> createState() {
+    return GenericExpansionCardState();
+  }
+
+  String getTitle();
+  Widget buildCardContent(BuildContext context);
+}
+
+class GenericExpansionCardState extends State<GenericExpansionCard> {
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+        margin: const EdgeInsets.symmetric(vertical: 10, horizontal: 20),
+        child: ExpansionTileCard(
+          expandedTextColor: Theme.of(context).primaryColor,
```

Comment:
The default heightFactorCurve is an easeOut. That curve looks like a little lag happens. I change it to ease and looks a lot better to me.

The turnsCurve is the animation curve of the arrow. I was playing with it and found a funny one as suggested on the bottom.

```suggestion
          expandedTextColor: Theme.of(context).primaryColor,
          heightFactorCurve: Curves.ease,
          turnsCurve: Curves.easeOutBack,
```

Reply:
hmm yeah, it's a funny detail in the icon's animation